common/type_system/TypeSystem.cpp

```diff
@@ -1304,6 +1304,138 @@ int TypeSystem::get_size_in_type(const Field& field) const {
   }
 }
 
+std::vector<std::string> TypeSystem::search_types_by_parent_type(
+    const std::string& parent_type,
+    const std::vector<std::string>& existing_matches) {
+  std::vector<std::string> results = {};
+  // If we've been given a list of already matched types, narrow it down from there, otherwise
+  // iterate through the entire map
+  if (!existing_matches.empty()) {
+    for (const auto& type_name : existing_matches) {
+      try {
+        if (is_type_descendent_from(type_name, parent_type)) {
+          results.push_back(type_name);
+        }
+      } catch (std::exception& e) {
+      }
+    }
+  } else {
+    for (const auto& [type_name, type_info] : m_types) {
+      try {
+        if (is_type_descendent_from(type_name, parent_type)) {
```

Comment:
This function might be doing the same thing as `typecheck_base_types(parent_type, type_name)`.